src/main/java/games/strategy/persistence/serializable/DefaultProxyFactoryRegistry.java

```diff
@@ -0,0 +1,44 @@
+package games.strategy.persistence.serializable;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+final class DefaultProxyFactoryRegistry implements ProxyFactoryRegistry {
+  private final Map<Class<?>, ProxyFactory> proxyFactoriesByPrincipalType = new HashMap<>();
+
+  @Override
+  public Optional<ProxyFactory> getProxyFactory(final Class<?> principalType) {
```

Comment:
I'd like to double check the usage of an `Optional` return type. My thoughts are we likely should be able to return a non-optional type. Looking at the client usage, it does seem like if we call this method and get an optional.empty() back,l it is more a problem with the client. That means if the client makes this call and gets an empty bag, it is perhaps more likely to be a bug rather than anything intentional. Fail fast principle would have us fail at that time rather than returning the empty.

Alternatively perhaps, we may want to update the API to have a `hasProxyFactory` method, so we don't mix the "discovery" and "retrieval" concerns.